web3/_utils/request.py

```diff
@@ -47,10 +47,18 @@ def get_default_http_endpoint() -> URI:
 
 
 def cache_and_return_session(
-    endpoint_uri: URI, session: requests.Session = None
+    endpoint_uri: URI, provider_id: int = None, session: requests.Session = None
```

Comment:
If we make `provider_id` a positional argument here and keyword argument everywhere else, we can "require" a `provider_id` each time this method is used. I think this is ideal if we remove the conditional logic below. This means closing the door on possible session sharing. Since this is an internal method, we could then make sure every time this is called we have a `provider_id`.

I think the only time this deviates from ideal is when the `handle_offchain_lookup()` utility methods are called, since those are public. I don't think those methods are ideal in their current form after this addition. It would be good to revisit those methods and see if there's a better flow there.